pkg/commands/compute/setup/config_store.go

```diff
@@ -113,91 +147,73 @@ func (o *ConfigStores) Create() error {
 		}
 	}
 
-	for _, store := range o.required {
-		err := o.Spinner.Start()
-		if err != nil {
-			return err
-		}
-		msg := fmt.Sprintf("Creating config store '%s'", store.Name)
-		o.Spinner.Message(msg + "...")
-
-		cs, err := o.APIClient.CreateConfigStore(&fastly.CreateConfigStoreInput{
-			Name: store.Name,
-		})
-		if err != nil {
-			err = fmt.Errorf("error creating config store: %w", err)
-			o.Spinner.StopFailMessage(msg)
-			spinErr := o.Spinner.StopFail()
-			if spinErr != nil {
-				return fmt.Errorf(text.SpinnerErrWrapper, spinErr, err)
-			}
-			return err
-		}
-
-		o.Spinner.StopMessage(msg)
-		err = o.Spinner.Stop()
-		if err != nil {
-			return err
-		}
+	for _, configStore := range o.required {
+		var (
+			err error
+			cs  *fastly.ConfigStore
+		)
 
-		if len(store.Items) > 0 {
-			for _, item := range store.Items {
-				err := o.Spinner.Start()
+		if configStore.LinkExistingStore {
+			err = o.Spinner.Process(fmt.Sprintf("Retrieving existing Config Store '%s'", configStore.Name), func(_ *text.SpinnerWrapper) error {
+				cs, err = o.APIClient.GetConfigStore(&fastly.GetConfigStoreInput{
+					ID: configStore.ExistingStoreID,
+				})
 				if err != nil {
-					return err
+					return fmt.Errorf("failed to get existing store '%s': %w", configStore.Name, err)
 				}
-				msg := fmt.Sprintf("Creating config store item '%s'", item.Key)
-				o.Spinner.Message(msg + "...")
-
-				_, err = o.APIClient.CreateConfigStoreItem(&fastly.CreateConfigStoreItemInput{
-					StoreID: cs.ID,
-					Key:     item.Key,
-					Value:   item.Value,
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+		} else {
+			err = o.Spinner.Process(fmt.Sprintf("Creating config store '%s'", configStore.Name), func(_ *text.SpinnerWrapper) error {
+				cs, err = o.APIClient.CreateConfigStore(&fastly.CreateConfigStoreInput{
+					Name: configStore.Name,
 				})
 				if err != nil {
-					err = fmt.Errorf("error creating config store item: %w", err)
-					o.Spinner.StopFailMessage(msg)
-					spinErr := o.Spinner.StopFail()
-					if spinErr != nil {
-						return fmt.Errorf(text.SpinnerErrWrapper, spinErr, err)
-					}
-					return err
+					return fmt.Errorf("error creating config store: %w", err)
 				}
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+		}
 
-				o.Spinner.StopMessage(msg)
-				err = o.Spinner.Stop()
+		if len(configStore.Items) > 0 {
+			for _, item := range configStore.Items {
+				err = o.Spinner.Process(fmt.Sprintf("Creating config store item '%s'", item.Key), func(_ *text.SpinnerWrapper) error {
+					_, err = o.APIClient.UpdateConfigStoreItem(&fastly.UpdateConfigStoreItemInput{
+						Upsert:  true, // Use upsert to avoid conflicts when reusing a starter kit.
```

Comment:
Is this documented somewhere- could cause some confusion if data keeps being overwritten.

Reply:
Hmm, good point actually. I might stick in a warning notice to let the user know we'll be upserting.